util/state.js

```diff
@@ -67,9 +67,17 @@ export class RedisCrawlState
     redis.defineCommand("addqueue", {
       numberOfKeys: 3,
       lua: `
-redis.call('sadd', KEYS[3], ARGV[1]);
+local size = redis.call('scard', KEYS[3]);
+local limit = tonumber(ARGV[4]);
+if limit > 0 and size >= limit then
+  return 0;
+end
+if redis.call('sadd', KEYS[3], ARGV[1]) == 0 then
+  return 0;
```

Comment:
This is causing the limit to be set as hit prematurely if we encounter a duplicate URL

Reply:
Fixed! Added different return values